### PR TITLE
Move graal build into build.yml

### PR DESCRIPTION
**Pull request by jack-berg**
Not clear why we should have a separate workflow for the `build-graal.yml`. Merged into `build.yml`.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -128,10 +128,29 @@ jobs:
...
+          java --version
+          gu --version
+          native-image --version
+          ./gradlew nativeTest
+
   required-status-check:
     # markdown-link-check is not required so pull requests are not blocked if external links break
     # misspell-check is not required so pull requests are not blocked if the misspell dictionary is
     # updated
+    # build-graal is not required so pull requests are not blocked if we're rate limited
```
Comment (trask) [inline .github/workflows/build.yml:153]: rate limited?

Comment (jack-berg) [reply .github/workflows/build.yml:153]: Errors like [this](https://github.com/open-telemetry/opentelemetry-java/actions/runs/3848150443/jobs/6555539275) happen fairly frequently.

Comment (jack-berg) [reply .github/workflows/build.yml:153]: Looks like this is a github rate limit, not something related to graal resources.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -128,10 +128,29 @@ jobs:
...
 
+  build-graal:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: graalvm/setup-graalvm@v1
+        with:
+          version: 'latest'
+          java-version: '11'
+          components: 'native-image'
```
Comment (trask) [inline .github/workflows/build.yml:139]: I think this should fix the rate limit errors:

```suggestion
          components: 'native-image'
          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment (jack-berg) [reply .github/workflows/build.yml:139]: Thanks!

**Review by trask — APPROVED** [review]
👍

**Pull request merged**